src/include/mpir_hw_topo.h

```diff
@@ -133,6 +133,11 @@ MPIR_Node_obj MPIR_Node_get_covering_obj_by_depth(int depth);
  */
 MPIR_Node_obj MPIR_Node_get_parent_obj(MPIR_Node_obj obj);
 
+/*
+ * Return object of type and index
+ */
+MPIR_Node_obj MPIR_Node_get_obj_by_type(MPIR_Node_obj_type obj_type, int obj_index);
```

Comment:
I think if we mean by type and index, then we should say `by_type_and_index` :) `hwloc` is sloppy in this, we don't have to follow suit. In fact, I would prefer `MPIR_Node_get_obj` since the type and index is clear in the parameters. (Not the `_Node_` part).

Reply:
The reason they have `_by_type` in `hwloc` is that you can also query `by_depth` in the topology tree. We could have had a overloaded `MPIR_Node_get_obj` in C++.